root/static/scripts/guess-case/modes.js

```diff
@@ -282,3 +291,24 @@ exports.Sentence = assign({}, DefaultMode, {
     {url: {href: 'https://musicbrainz.org/doc/Style/Language', target: '_blank'}}
   ),
 });
+
+exports.Turkish = assign({}, DefaultMode, {
+  description: l(
+    'This mode handles the Turkish capitalisation of \'i\' (\'İ\') and \'ı\' '+
+    '(\'I\'). Some words may need to be manually corrected '+
+    'according to the {url|Turkish language guidelines}. ',
+    {url: {href: 'https://musicbrainz.org/doc/Style/Language/Turkish', target: '_blank'}}
+  ),
+
+  isSentenceCaps() {
+    return false;
+  },
+
+  toUpperCase(str) {
+    return turkishUpperCase(str);
+  },
+
+  toLowerCase(str) {
+    return turkishLowerCase(str);
+  },
```

Comment:
Another thing I noticed is that it isn’t necessary to have these function definitions. Since the signatures match, it should be enough to write `toUpperCase: turkishUpperCase, toLowerCase: turkishLowerCase`.